pytorch_lightning/utilities/cli.py

```diff
@@ -602,23 +535,18 @@ def add_core_arguments_to_parser(self, parser: LightningArgumentParser) -> None:
         """Adds arguments from the core classes to the parser."""
         parser.add_lightning_class_args(self.trainer_class, "trainer")
         parser.set_choices("trainer.callbacks", CALLBACK_REGISTRY.classes, is_list=True)
```

Comment:
> > Until _convert_argv_issue_85 is removed, still there is a need to register the callbacks. But assuming that this is done, there is a general question. Should the registries be deprecated?
> 
> If we have no use for them, yes.

Additional things I could do in this pull request are:
- Remove `LightningArgumentParser.set_choices` and this only remaining use of it.
- In `LightningArgumentParser.parse_args` instead of `self._choices.items()` use jsonargparse's function to list known subclasses and call `_convert_argv_issue_85` if needed.
- Deprecate the registries and update the documentation.

Does this sound good or should this be a different pull request?

Reply:
I'd leave "Deprecate the registries and update the documentation" for another PR, the rest are internal and can be done here

Reply:
Done.